src/WebServer.ino

```diff
@@ -300,7 +300,10 @@ void sendWebPageChunkedBegin(String& log)
 
 void sendWebPageChunkedData(String& log, String& data)
 {
-  while (ESP.getFreeHeap() < 7000) backgroundtasks(); 
+  long timeout= millis();  
+  while ((ESP.getFreeHeap() < 7000) && ((millis()-timeout < 100)) ) {
```

Comment:
Use something like this to wait for timeout:
```
uint32_t beginWait = millis();
while (!timeOutReached(beginWait + 1000)) {
```
It is better readable and does not cause issues when millis() wrap.